crates/languages/src/cpp/highlights.scm

```diff
@@ -64,6 +68,14 @@
 
 (auto) @type
 (type_identifier) @type
+type :(primitive_type) @type.primitive
```

Comment:
For this commit I used [clangd's tokens](https://clangd.llvm.org/features#semantic-highlighting) as a reference, I also wanted to keep the possibility of highlighting primitives types differently than user defined types so I specialized the tag.

I think changing type.primitive to type.builtin is a good idea to keep things consistent.

It would also be good to have more documentation regarding syntax highlighting tags, a list of the available tags is difficult to find and would have its place on [this page](https://zed.dev/docs/extensions/themes).

Reply:
> It would also be good to have more documentation regarding syntax highlighting tags, a list of the available tags is difficult to find and would have its place on [this page](https://zed.dev/docs/extensions/themes).

That would be excellent (happy to review PRs!).  Similarly we should include a list of default syntax types we expect a theme to provide colors for as part of the [theme json schema](https://zed.dev/schema/themes/v0.1.0.json) currently we only enforce the schema of those objects but we could include a list of `required` keys to easily help theme creators discover the syntax items they are missing.